// @clerk/nextjs/dist/cjs/server/routeMatcher.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var routeMatcher_exports = {};
__export(routeMatcher_exports, {
  MalformedURLError: () => MalformedURLError,
  createRouteMatcher: () => createRouteMatcher,
  isMalformedURLError: () => isMalformedURLError
});
module.exports = __toCommonJS(routeMatcher_exports);
var import_pathToRegexp = require("@clerk/shared/pathToRegexp");
class MalformedURLError extends Error {
  constructor(pathname, cause) {
    super(`Malformed encoding in URL path: ${pathname}`);
    this.statusCode = 400;
    this.name = "MalformedURLError";
    this.cause = cause;
  }
}
function isMalformedURLError(e) {
  return e instanceof Error && e.name === "MalformedURLError";
}
const normalizePath = (pathname) => {
  try {
    pathname = decodeURI(pathname);
  } catch (e) {
    throw new MalformedURLError(pathname, e);
  }
  return pathname.replace(/\/\/+/g, "/");
};
const createRouteMatcher = (routes) => {
  if (typeof routes === "function") {
    return (req) => routes(req);
  }
  const routePatterns = [routes || ""].flat().filter(Boolean);
  const matchers = precomputePathRegex(routePatterns);
  return (req) => matchers.some((matcher) => matcher.test(normalizePath(req.nextUrl.pathname)));
};
const precomputePathRegex = (patterns) => {
  return patterns.map((pattern) => pattern instanceof RegExp ? pattern : (0, import_pathToRegexp.pathToRegexp)(pattern));
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  MalformedURLError,
  createRouteMatcher,
  isMalformedURLError
});
