// @clerk/nextjs/dist/esm/server/routeMatcher.js
import { pathToRegexp } from "@clerk/shared/pathToRegexp";
class MalformedURLError extends Error {
  constructor(pathname, cause) {
    super(`Malformed encoding in URL path: ${pathname}`);
    this.statusCode = 400;
    this.name = "MalformedURLError";
    this.cause = cause;
  }
}
function isMalformedURLError(e) {
  return e instanceof Error && e.name === "MalformedURLError";
}
const normalizePath = (pathname) => {
  try {
    pathname = decodeURI(pathname);
  } catch (e) {
    throw new MalformedURLError(pathname, e);
  }
  return pathname.replace(/\/\/+/g, "/");
};
const createRouteMatcher = (routes) => {
  if (typeof routes === "function") {
    return (req) => routes(req);
  }
  const routePatterns = [routes || ""].flat().filter(Boolean);
  const matchers = precomputePathRegex(routePatterns);
  return (req) => matchers.some((matcher) => matcher.test(normalizePath(req.nextUrl.pathname)));
};
const precomputePathRegex = (patterns) => {
  return patterns.map((pattern) => pattern instanceof RegExp ? pattern : pathToRegexp(pattern));
};
export {
  MalformedURLError,
  createRouteMatcher,
  isMalformedURLError
};
